crt_portal/cts_forms/views.py

```diff
@@ -448,6 +452,101 @@ def post(self, request, id):
             return render(request, 'forms/complaint_view/show/index.html', output)
 
 
+class ActionsView(LoginRequiredMixin, FormView):
+
+    def reconstruct_query(self, next_qp):
+        """
+        reconstruct the query filter on the previous page using the next
+        query parameter. note that if next is empty, the resulting
+        query will return all records.
+        """
+        querydict = QueryDict(next_qp)
+        report_query, _ = report_filter(querydict)
+        sort = querydict.getlist('sort', ['-create_date'])
+        return report_query.order_by(*sort)
+
+    def get(self, request):
+        return_url_args = request.GET.get('next', '')
+        return_url_args = urllib.parse.unquote(return_url_args)
+
+        requested_query = self.reconstruct_query(return_url_args)
+        all_ids_count = requested_query.count()
+
+        ids = request.GET.getlist('id')
+        ids_count = len(ids)
+        assign_form = BulkAssign()
+
+        # the select all option only applies if 1. user hits the
+        # select all button and 2. we have more records in the query
+        # than the ids passed in
+        selected_all = request.GET.get('all', '') == 'all' and all_ids_count != ids_count
+
+        output = {
+            'return_url_args': return_url_args,
+            'selected_all': 'all' if selected_all else '',
+            'ids': ','.join([id for id in ids]),
+            'ids_count': ids_count,
+            'all_ids_count': all_ids_count,
+            'assign_form': assign_form,
+        }
+        return render(request, 'forms/complaint_view/actions/index.html', output)
+
+    def post(self, request):
+        assign_form = BulkAssign(request.POST)
+        return_url_args = request.POST.get('next', '')
+        selected_all = request.POST.get('all', '') == 'all'
+        ids = request.POST.get('ids', '').split(',')
+
+        if assign_form.is_valid():
+            assignee = assign_form.cleaned_data['assigned_to']
+            requested_query = None
```

Comment:
This doesn't look necessary since we're assigning a value in any situation immediately thereafter.

Reply:
Thanks! This was a leftover from when it was a large if then statement. Fixed.